hazelcast/src/main/java/com/hazelcast/instance/NodeExtension.java

```diff
@@ -33,6 +34,7 @@
  * NodeExtension is a <tt>Node</tt> extension mechanism to be able to plug different implementations of
  * some modules, like; <tt>SerializationService</tt>, <tt>SocketChannelWrapperFactory</tt> etc.
  */
+@PrivateApi
 public interface NodeExtension {
```

Comment:
Reading the Javadoc it sounds to me like public API tbh and I know at least one project that uses this, @jerrinot :)